typhos/display.py

```diff
@@ -1323,10 +1329,18 @@ def _load_template(self, filename):
         """Load template from file and return the widget."""
         filename = pathlib.Path(filename)
         loader = (pydm.display.load_py_file if filename.suffix == '.py'
-                  else pydm.display.load_ui_file)
+                  else utils.load_ui_file)
 
         logger.debug('Load template using %s: %r', loader.__name__, filename)
-        return loader(str(filename), macros=self._macros)
+        try:
+            return loader(str(filename), macros=self._macros)
+        except Exception as ex:
+            display: Optional[pydm.Display] = getattr(ex, "pydm_display", None)
+            if display is not None:
+                display.setObjectName("_typhos_test_suite_ignore_")
```

Comment:
Functionally, setting the object name for a `pydm.Display` does nothing - it's a big of a magic string for the test suite and for us when looking in with debug tools is all.

The context of this block is:
* The new template fails to load
* typhos notices that and tries to reload the prior template

In typhos master,
* A `pydm.Display` is left dangling as a top-level widget and `TyphosDeviceDisplay` never even sees a reference to it

In this PR,
* That `pydm.Display` is seen by `TyphosDeviceDisplay` and intercepted as a failed load
* We mark it for deletion and wave goodbye
* The test suite *still* picks it up because it's a top-level widget and somehow doesn't get destroyed in time for it to be happy (*see below*)
* The test suite sees the "magic" object name and says "OK this is known and expected; let's not fail the test"

Now to be honest, the marked step is fishy and it'd be nice to understand why that's happening. Is there a reference we're missing? It's worth investigating at some point.